### PR TITLE
Fix getting a read timeout for logs/attach with a tty and slow output

### LINKED ISSUE #931: getting timeout after streaming logs for some time
This is very likely an engine issue, first I would like to open here.

It looks like there is a change in behavior. When streaming logs with 1.10 I get timeout:

```
Traceback (most recent call last):
  File "/usr/lib64/python3.5/threading.py", line 914, in _bootstrap_inner
    self.run()
  File "/usr/lib64/python3.5/threading.py", line 862, in run
    self._target(*self._args, **self._kwargs)
  File "/home/tt/g/sen/sen/tui/widgets/list/common.py", line 43, in fetch_logs
    for line in generator:
  File "/home/tt/.local/lib/python3.5/site-packages/docker/client.py", line 279, in _stream_raw_result
    for out in response.iter_content(chunk_size=1, decode_unicode=True):
  File "/usr/lib/python3.5/site-packages/requests/utils.py", line 363, in stream_decode_response_unicode
    for chunk in iterator:
  File "/usr/lib/python3.5/site-packages/requests/models.py", line 667, in generate
    raise ConnectionError(e)
requests.exceptions.ConnectionError: UnixHTTPConnectionPool(host='localhost', port=None): Read timed out.
```

generator is `client.logs(c, stream=True)`

This repro should be sufficient: https://github.com/docker/docker-py/issues/919#issuecomment-181795059

I know this also happens when using compose.

### PR DESCRIPTION
**Pull request by segevfiner**
Fixes #931

P.S. There is another possible instance of getting an unexpected timeout in `attach_socket` since you return the socket from requests/urllib3 to the user, which will have a timeout set.

@shin-

### CODE REVIEW

**Comment by akki** [comment]
Would any of the maintainers like to review this PR?

What would help in resolving https://github.com/docker/docker-py/issues/931?

**Comment by akki** [comment]
@TomasTomecek Since you created https://github.com/docker/docker-py/issues/931 and you find this PR to be its solution for you to merge it? Or can you tell what else we can do for it to get into `master`?

**Comment by akki** [comment]
We are affected in Airflow because of this issue. Although there are ways [to hack around it](https://github.com/akki/airflow/commit/b008c26e526f1b2fbb9e0d95016241f8e26b7c10#diff-6cba2b9256f3d84f52b9f9e3dcb72439R143) but we would love it if we could fix this properly.

**Comment by TomasTomecek** [comment]
> @TomasTomecek Since you created #931 and you find this PR to be its solution for you to merge it? Or can you tell what else we can do for it to get into `master`?

Find a maintainer of this repository and work with the person on a review for this PR. I'm not a maintainer, only a contributor.

Since @shin- already assigned this PR to himself, hopefully you'd get a maintainer review from him soon.

**Comment by CodingJonas** [comment]
@shin-, I hope it is okay to ping you. 

This MR is open for over two years now, what is blocking it? I would be happy to help in any way to get this merged.

**Comment by HaloKo4** [comment]
@shin- is there a chance you can review?

**Comment by akki** [comment]
This PR is near its 3 years anniversary without any activity from any repo maintainers. So I hope it is okay to ping @ulyssessouza @aiordache to get their attention as they seem to be the most active maintainers of this repo right now.

**Comment by michaelfresco** [comment]
This PR would be so cool for Airflow and Docker Swarm. I hope this PR can be approved. Small ping to @ulyssessouza, @aiordache, @thaJeztah

**Comment by HaloKo4** [comment]
hey @aiordache @ulyssessouza

could you please get some attention to this PR? :) 
3+ years for 2 lines for code is not reasonable.
I don't know why Docker ignores this PR - you are making you customers life really difficult.

**Comment by HaloKo4** [comment]
@ulyssessouza  @aiordache @thaJeztah 
Kindly ask for a comment, any comment. 
What else are we to do to get attention from the repo maintainers? :(

**Comment by thaJeztah** [comment]
I'm not a maintainer for this repository (and not very proficient on Python code), but perhaps one of the others can have a look @ulyssessouza @aiordache 

Looks like this PR needs a rebase though (I see there's a merge conflict)

**Comment by HaloKo4** [comment]
> Looks like this PR needs a rebase though (I see there's a merge conflict)

It's 2 lines of code. Assuming this PR is accepted I have no problem to open a new PR with it or the maintainers can take over. This is really small modification. If not accepted kindly explain how maintainers suggest to solve the problem.

**Comment by thaJeztah** [comment]
> It's 2 lines of code.

I understand it's a small change, and I see a similar approach is taken in other parts of this file, but it still needs to be reviewed. A single line change does not make [a change less risky](https://nakedsecurity.sophos.com/2014/02/24/anatomy-of-a-goto-fail-apples-ssl-bug-explained-plus-an-unofficial-patch/).

As said; I'm not a Python coder, so I defer review to others mentioned. It would be good if @segevfiner could rebase this PR and resolve the merge conflict, otherwise it can't be merged.

**Review by ulyssessouza — APPROVED** [review]
LGTM. Just needs a rebase.

**Comment by segevfiner** [comment]
Rebased.

**Review by aiordache — APPROVED** [review]

**Comment by HaloKo4** [comment]
@aiordache when should we expect 5.0.3 to be released?

**Comment by aiordache** [comment]
@HaloKo4 Sorry for the delay, we'll have 5.0.3 out by Oct. 8th

**Pull request merged**
